wp_api/src/post_types.rs

```diff
@@ -36,11 +38,40 @@ impl Display for PostType {
             Self::WpFontFamily => "wp_font_family",
             Self::WpFontFace => "wp_font_face",
             Self::Custom(name) => name.as_str(),
-        };
-        write!(f, "{}", s)
+        }
+    }
+}
+
+impl FromStr for PostType {
+    type Err = EnumFromStrParsingError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "post" => Ok(Self::Post),
+            "page" => Ok(Self::Page),
+            "attachment" => Ok(Self::Attachment),
+            "nav_menu_item" => Ok(Self::NavMenuItem),
+            "wp_block" => Ok(Self::WpBlock),
+            "wp_template" => Ok(Self::WpTemplate),
+            "wp_template_part" => Ok(Self::WpTemplatePart),
+            "wp_navigation" => Ok(Self::WpNavigation),
+            "wp_font_family" => Ok(Self::WpFontFamily),
+            "wp_font_face" => Ok(Self::WpFontFace),
+            value => Err(EnumFromStrParsingError::UnknownVariant {
+                value: value.to_string(),
+            }),
```

Comment:
Should this case return `PostType::Custom(value)`? IMHO, it's more ideal if this conversion between concrete type and string works both ways. For example:

```rust
let type: PostType = PostType::Custom("foo")
assert_eq!(type.as_str(), "foo")

let converted: PostType = type.as_str().into()
assert_eq!(type, converted)
```

Reply:
It definitely should be, thanks for catching that! Addressed in 0afb76c1ff8486a0d3a10133f45d1c5d01fc08d3.

Hopefully I didn't make the same mistake in the other PRs that build on this 😓

Reply:
We need to implement `FromStr` because of [this trait](https://github.com/Automattic/wordpress-rs/blob/trunk/wp_api/src/url_query.rs#L108-L110) bound. Some of the types can't be converted from `String`, so we have to be able to throw an error.

For this type, we could implement `From<&str>` and then call that in the `FromStr` implementation, but that's currently not necessary. If it becomes necessary, we can do that for all the available types in a single PR, rather than break the established pattern in this one.

Reply:
Maybe changing the error type to `std::convert::Infallible`?

Reply:
TIL!

I've made the change in 42eedcbddda31ef6763e5f736c098892d7adb0cd - including for some other types that would benefit from this.